src/odin/narrativebuilder.cc

```diff
@@ -1163,6 +1196,20 @@ std::string NarrativeBuilder::FormTurnInstruction(Maneuver& maneuver,
     phrase_id = 1;
   }
 
+  // Select the most suitable landmark if there are landmarks associated with the maneuver
+  if (phrase_id < 4 && maneuver.landmarks().size() > 0) {
+    landmark =
+        select_landmark(maneuver.landmarks(),
+                        FormRelativeTwoDirection(maneuver.type(), subset->relative_directions));
+
+    // Make sure the landmark type is legal
+    if (static_cast<int>(landmark.type()) > 0 &&
+        static_cast<int>(landmark.type()) < subset->landmark_types.size()) {
+      // increment the phrase id by 6 to get the corresponding landmark phrase
+      phrase_id += 6;
+    }
```

Comment:
IMO we don't need a check like this, bcs we're in full control of the landmark types and we do make sure in the data building pipeline that nothing enters that's outside of the type bounds. If we want to check this at this point, then let's throw an exception, bcs that'd be a very serious flaw.